src/main/controllers/AgreeingDocumentsForHearingController.ts

```diff
@@ -0,0 +1,130 @@
+                id: 'bundlesRespondentAgreedDocWithBut',
+                name: 'bundlesRespondentAgreedDocWithBut',
+                type: 'textarea',
+                hint: (l: AnyRecord): string => l.agreedTextLabel,
+                classes: 'govuk-textarea',
+                attributes: { maxLength: 2500 },
+                validator: isFieldFilledIn,
+              },
+            },
+          },
+          {
+            name: 'bundlesRespondentAgreedDocWith',
+            label: (l: AnyRecord): string => l.notAgreed,
+            value: AgreedDocuments.NOTAGREED,
+            subFields: {
+              bundlesRespondentAgreedDocWithNo: {
+                id: 'bundlesRespondentAgreedDocWithNo',
+                name: 'bundlesRespondentAgreedDocWithNo',
+                type: 'textarea',
+                hint: (l: AnyRecord): string => l.notAgreedTextLabel,
+                classes: 'govuk-textarea',
+                attributes: { maxLength: 2500 },
+                validator: isFieldFilledIn,
+              },
+            },
+          },
+        ],
+        validator: isFieldFilledIn,
+      },
+    },
+    submit: {
+      text: l => l.continue,
+    },
+  };
+
+  constructor() {
+    this.form = new Form(<FormFields>this.agreeingDocumentsForHearingContent.fields);
+  }
+
+  public post = async (req: AppRequest, res: Response): Promise<void> => {
+    const { userCase } = req.session;
+    req.session.errors = [];
+
+    const radios = req.body.bundlesRespondentAgreedDocWith;
+    let agreedButText = req.body.bundlesRespondentAgreedDocWithBut;
+    let notAgreedText = req.body.bundlesRespondentAgreedDocWithNo;
+
+    if (radios === AgreedDocuments.YES) {
+      userCase.bundlesRespondentAgreedDocWith = radios;
+      userCase.bundlesRespondentAgreedDocWithBut = undefined;
+      userCase.bundlesRespondentAgreedDocWithNo = undefined;
+      agreedButText = undefined;
+      notAgreedText = undefined;
+    }
+
+    if (radios === AgreedDocuments.AGREEDBUT && agreedButText !== undefined) {
+      userCase.bundlesRespondentAgreedDocWith = radios;
+      userCase.bundlesRespondentAgreedDocWithBut = agreedButText;
+      userCase.bundlesRespondentAgreedDocWithNo = undefined;
+      notAgreedText = undefined;
+    }
+
+    if (radios === AgreedDocuments.NOTAGREED && notAgreedText !== undefined) {
+      userCase.bundlesRespondentAgreedDocWith = radios;
+      userCase.bundlesRespondentAgreedDocWithNo = notAgreedText;
+      userCase.bundlesRespondentAgreedDocWithBut = undefined;
+      agreedButText = undefined;
+    }
+
+    req.session.errors = agreeingDocumentsForHearingErrors(req);
+    if (req.session?.errors?.length) {
+      return res.redirect(PageUrls.AGREEING_DOCUMENTS_FOR_HEARING);
+    }
+    req.session.errors = [];
+    return res.redirect(PageUrls.ABOUT_HEARING_DOCUMENTS);
+  };
+
+  public get = (req: AppRequest, res: Response): void => {
+    const userCase = req.session?.userCase;
+    console.log(userCase);
```

Comment:
We want the console.log here? Check if there are any other ones.

Reply:
Removed